# RotationProject2_BacterialGWAS
Rotation Project 2: Bacterial Genome Wide Association Study (GWAS)
31/07/2022
Hayley Free - BBSRC Oxford Interdisciplinary Biosicences DTP
Supervisor: Dr Daniel Wilson
Big Data Institute, Old Road Campus

This is the R code and associated files used for the multiple parts of analysis for bacterial GWAS for Campylobacter coli host specificity to humans, chickens and pig sources. This is for the completion of the second rotation project.

This project involved using the R package bacterial GWAS - bugwas - developed by Earle, S.G. et al. (2016) “Identifying lineage effects when controlling for population structure improves power in bacterial association studies,” Nature microbiology, 1(5), p. 16041. 
This was used on data obtained from the publicly available sources at PubMLST. It was undertaken to investigate the host specificity of C. coli as an important source of bacterial gastroenteritis in humans. This therefore aims to enable the analysis of causal variants to add to current findings. 

This folder for the project requires additional downloads to reproduce the study (such as Mash to create the distance matrix and miniconda3 to run the Mash). Within this repository is the code written for the various parts of the analysis undertaken in RStudio, and the accompanying files (e.g. downloaded metadata from PubMLST, Mash distance matrix…). This does not include the raw data.
